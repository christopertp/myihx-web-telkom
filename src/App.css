* {
  animation: fade-in 0.5s ease-in-out;
  box-sizing: border-box;
  color: inherit;
  font-family: 'Ubuntu', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
  line-height: 1.5;
  list-style: none;
  margin: 0;
  padding: 0;
}

@keyframes loading {
  to { transform: translateX(100%); }
}

@keyframes fade-in {
  from { opacity: 0; }
  to { opacity: 1; }
}

a {
  color: inherit;
  text-decoration: none;
}

body {
  overflow-x: hidden;
}

#app {
  min-height: 100vh;
  overflow-x: hidden;
  position: relative;
  width: 100vw;
}

#app.home {
  display: flex;
  flex-direction: column;
}

#app.home > main > *:nth-child(1) { animation: fade-in 1s ease-in-out 0.25s; animation-fill-mode: forwards; opacity: 0; }
#app.home > main > *:nth-child(2) { animation: fade-in 1s ease-in-out 0.75s; animation-fill-mode: forwards; opacity: 0; }
#app.home > main > *:nth-child(3) { animation: fade-in 1s ease-in-out 1.25s; animation-fill-mode: forwards; opacity: 0; }
#app.home > main > *:nth-child(4) { animation: fade-in 1s ease-in-out 1.75s; animation-fill-mode: forwards; opacity: 0; }
#app.home > main > *:nth-child(5) { animation: fade-in 1s ease-in-out 2.25s; animation-fill-mode: forwards; opacity: 0; }

body.white-card {
  background-color: #f2f2f2;
  min-height: 100vh;
}

body.white-card > #app::after {
  background: url(/assets/bg_header_white_full.png) no-repeat;
  background-size: cover;
  content: '';
  height: 25rem;
  left: 0;
  position: absolute;
  top: 3.75rem;
  width: 100%;
  z-index: -1;
}

.loading {
  background-color: #dddbdd;
  position: relative;
  overflow: hidden;
}

.loading::after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  transform: translateX(-100%);
  background-image: linear-gradient(90deg, 
    rgba(255, 255, 255, 0) 0, 
    rgba(255, 255, 255, 0.3) 50%,
    rgba(255, 255, 255, 0));
  animation: loading 2s infinite;
  content: '';
}

.modal-open {
  overflow: hidden;
}

.modal-open > div > .modal {
  background-color: rgba(0, 0, 0, 0.6);
  display: block;
}

.modal {
  display: none;
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  transition: background-color 0.25s;
  width: 100vw;
  z-index: 101;
}

.notif {
  background-color: #0eb37e;
  box-shadow: 0 0.25rem 0.625rem rgba(0, 0, 0, 0.25);
  border-radius: 0.5rem;
  color: #ffffff;
  padding: 1rem;
  position: fixed;
  right: 1rem;
  top: 1rem;
  transform: translateX(100vw);
  transition: transform 0.25s ease-in-out;
  width: 25rem;
  z-index: 103;
}

.notif.active {
  transform: translateX(0);
}

.overlay {
  background-color: rgba(0, 0, 0, 0.6);
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  transition: background-color 0.25s;
  width: 100vw;
  z-index: 102;
}

:root {
  --primary-color: #ee3124;
  --side-margin: calc(50vw - 35.75rem);
  --text-color: #191a1a;
  --slate-grey: #66686a;
  --neg-side-margin: calc(35.75rem - 50vw);
  font-size: 16px;
}

@media only screen and (max-width: 90em) {
  :root {
    --side-margin: 9.25rem;
    --neg-side-margin: -9.25rem;
    font-size: calc(100vw * 16 / 1440);
  }  
}

@media only screen and (max-width: 48em) {
  * {
    line-height: 1.2;
  }

  :root {
    font-size: calc(100vw * 16 / 360);
  }

  body.white-card {
    background-color: #ffffff;
    min-height: auto;
  }

  body.white-card > #app::after {
    content: none;
  }

  .notif {
    width: calc(100% - 2rem);
  }
}
