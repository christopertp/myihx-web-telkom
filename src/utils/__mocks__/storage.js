export const setToken = jest.fn();
export const getToken = jest.fn(() => true);
export const clearStorages = jest.fn();
export const setExpireTime = jest.fn();
export const checkExpireTime = jest.fn(() => false);
export const setUserData = jest.fn();
export const getUserData = jest.fn(() => ({ profilePicture: 'tes' }));
export const setGuestAddress = jest.fn();
export const getGuestAddress = jest.fn(() => '');
export const clearGuestAddress = jest.fn();
