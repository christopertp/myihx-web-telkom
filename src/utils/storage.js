const EXPIRE_TIME_STORAGE = 'myihx_web_expire_time';
const GUEST_ADDRESS_STORAGE = 'myihx_web_guest_address';
const SESSION_TIME = 'myihx_web_session_time';
const TOKEN_STORAGE = 'myihx_web_access_token';
const USER_DATA_STORAGE = 'myihx_web_user_data';

export function setToken(value) {
  localStorage.setItem(TOKEN_STORAGE, value);
}

export function getToken() {
  return localStorage.getItem(TOKEN_STORAGE);
}

export function clearStorages() {
  localStorage.removeItem(TOKEN_STORAGE);
  localStorage.removeItem(EXPIRE_TIME_STORAGE);
  localStorage.removeItem(USER_DATA_STORAGE);
  localStorage.removeItem(GUEST_ADDRESS_STORAGE);
  localStorage.removeItem(SESSION_TIME);
}

export function setExpireTime(value) {
  localStorage.setItem(EXPIRE_TIME_STORAGE, value * 1000);
}

export function checkExpireTime() {
  const time = new Date().getTime();
  const expire = localStorage.getItem(EXPIRE_TIME_STORAGE) || 0;

  return time > expire;
}

export function setUserData(value) {
  localStorage.setItem(USER_DATA_STORAGE, JSON.stringify(value));
}

export function getUserData() {
  const retval = localStorage.getItem(USER_DATA_STORAGE);

  return JSON.parse(retval) || '';
}

export function setGuestAddress(value) {
  localStorage.setItem(GUEST_ADDRESS_STORAGE, JSON.stringify(value));
}

export function getGuestAddress() {
  const retval = localStorage.getItem(GUEST_ADDRESS_STORAGE);

  return JSON.parse(retval) || '';
}

export function clearGuestAddress() {
  localStorage.removeItem(GUEST_ADDRESS_STORAGE);
}

export function setSessionTime(value) {
  localStorage.setItem(SESSION_TIME, value);
}

export function getSessionTime() {
  const retval = localStorage.getItem(SESSION_TIME);

  return parseInt(retval) || 0;
}
