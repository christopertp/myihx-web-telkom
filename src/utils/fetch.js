import axios from 'axios';
import { getToken } from './storage';

const lng = 'en';
export const BASIC_AUTH = {
  'Authorization': 'Basic dGVsa29tOmRhMWMyNWQ4LTM3YzgtNDFiMS1hZmUyLTQyZGQ0ODI1YmZlYQ==',
  'Accept-Language': lng,
};
export const BEARER_AUTH = {
  'Authorization': `Bearer ${getToken()}`,
  'Accept-Language': lng,
};

export default function fetch(url, method, param1, param2) {
  return new Promise((resolve, reject) => {
    axios[method](url, param1, param2)
      .then(res => resolve(res.data))
      .catch(err => {
        const defaultError = {
          code: 500,
          status: 'error',
          message: 'Failed to fetch data. Please contact developer.'
        };

        if (!err.response) {
          reject(defaultError);
        } else if (!err.response.data) {
          reject(defaultError);
        } else {
          reject(err.response.data);
        }
      });
  });
}

export function getUrl(urls) {
  const mode = process.env.MODE;

  if (mode === 'production') {
    return urls[0];
  }

  if (mode === 'staging') {
    return urls[1];
  }

  return urls[2];
}
