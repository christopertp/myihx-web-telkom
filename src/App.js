import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader/root';
import { Router } from 'react-router';
import { Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import ErrorBoundary from './components/elements/ErrorBoundary';
import MainBase from './components/layouts/MainBase';
import pages from './pages';
import { checkExpireTime, clearStorages, getToken } from './utils/storage';
import ContextProvider from './contexts';

if (!('scrollBehavior' in document.documentElement.style)) {
  import('scroll-behavior-polyfill');
}

const afterLoginRoutes = [
  { c: pages.About, key: 'About', path: '/about', },
  { c: pages.ActivateRewards, key: 'ActivateRewards', path: ['/activate-rewards', '/activate-rewards/:page'], },
  { c: pages.Bill, key: 'Bill', path: '/bill', },
  { c: pages.CompleteProfile, key: 'CompleteProfile', path:  ['/complete-profile', '/complete-profile/:page'], },
  { c: pages.DraftContract, key: 'DraftContract', path: '/draft-contract', },
  { c: pages.History, key: 'History', path: ['/history', '/history/:page'], },
  { c: pages.Inbox, key: 'Inbox', path: ['/inbox', '/inbox/:page'], },
  { c: pages.Issues, key: 'Issues', path: ['/issues', '/issues/:id'], },
  { c: pages.LinkAja, key: 'LinkAja', path: ['/linkaja', '/linkaja/:page'], },
  { c: pages.ManageSubscriptions, key: 'ManageSubscriptions', path: ['/manage-subscriptions', '/manage-subscriptions/:page'], },
  { c: pages.ManageUsers, key: 'ManageUsers', path: ['/manage-users', '/manage-users/:id','/manage-users/:id/:action'], },
  { c: pages.NotificationSettings, key: 'NotificationSettings', path: ['/notification-settings'], },
  { c: pages.Installation, key: 'Installation', path: ['/installation', '/installation/:page'], },
  { c: pages.Order, key: 'Order', path: ['/order/:page', '/order/:page/:subpage'], },
  { c: pages.Payment, key: 'Payment', path: ['/payment-:code/:method', '/payment-:code/:method/:detail', '/payment-:code/:method/:detail/:bank'], },
  { c: pages.PersonalData, key: 'PersonalData', path: ['/personal-data', '/personal-data/:page', '/personal-data/:page/:subpage'], },
  { c: pages.Profile, key: 'Profile', path: ['/profile', '/profile/:page', '/profile/:page/:subpage'], },
  { c: pages.Promo, key: 'Promo', path: ['/promo/:name'], },
  { c: pages.Receipt, key: 'Receipt', path: '/receipt/:period', },
  { c: pages.Rewards, key: 'Rewards', path: ['/rewards/:page', '/rewards/:page/:id'], },
  { c: pages.ScheduleAppointment, key: 'ScheduleAppointment', path: ['/schedule-:type', '/schedule-:type/:page'], },
  { c: pages.SendFeedback, key: 'SendFeedback', path: '/send-feedback', },
  { c: pages.Tmoney, key: 'Tmoney', path: ['/tmoney', '/tmoney/:page'], },
];

const BeforeLoginRoutes = <Route component={pages.Register} exact key="Register" path={['/register', '/register/:page']} />;

export const MainPages = () => (
  <MainBase>
    <Switch>
      {afterLoginRoutes.map(i => (
        <Route component={getToken() ? i.c : pages.PleaseLogin} exact key={i.key} path={i.path} />
      ))}
      <Route component={pages.CheckCoverage} exact path={['/check-coverage', '/check-coverage/:page']} />
      <Route component={pages.DummyNps} exact path="/dummy-nps" />
      <Route component={pages.Help} exact path={['/help', '/help/:page']} />
      <Route component={pages.PackageCategory} exact path={['/shop/internet/package/category','/shop/internet/package/category/:category']} />
      <Route component={pages.PackageDetail} exact path="/shop/internet/package/:productId" />
      <Route component={pages.SearchProduct} exact path="/search" />
      <Route component={pages.ShopInternet} exact path={['/shop/internet', '/shop/internet/:page']} />
      <Route component={pages.ShopMore} exact path={['/shop/more', '/shop/more/:page']} />
      <Route component={pages.Home} exact path={['/', '/shop']} />
      <Route component={pages.Error404} />
    </Switch>
  </MainBase>
);

const App = ({ history, store }) => {
  if (getToken() && checkExpireTime()) {
    clearStorages();
    location.reload();
    return null;
  }
  return (
    <Provider store={store}>
      <Router history={history}>
        <ErrorBoundary>
          <ContextProvider>
            <Switch>
              {!getToken() ? BeforeLoginRoutes : null}
              <Route component={pages.ForgotPassword} exact key="ForgotPassword" path={['/forgot-password', '/forgot-password/:page']} />
              <Route component={pages.Login} exact key="Login" path={['/login', '/login/:page']} />
              <Route component={MainPages} />
            </Switch>
          </ContextProvider>
        </ErrorBoundary>
      </Router>
    </Provider>
  );
};

export default hot(App);

App.propTypes = {
  history: PropTypes.object.isRequired,
  store: PropTypes.object.isRequired,
};
