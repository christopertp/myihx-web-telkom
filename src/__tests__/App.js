import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { checkExpireTime, clearStorages, getToken } from '../utils/storage';
import App, { MainPages } from '../App';

jest.mock('react-hot-loader/root', () => ({ hot: c => c }));
jest.mock('../pages', () => ({ Login: 'login' }));
jest.mock('../utils/storage');

delete window.location;
window.location = { reload: jest.fn() };

describe('src/App', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    };
    const tree = shallow.render(<App history={{}} store={store} />);
    expect(tree).toMatchSnapshot();

    getToken.mockReturnValueOnce(false).mockReturnValueOnce(false);
    const result = App({ history: {}, store });
    const ErrorBoundary = result.props.children.props.children.props.children;
    const routes = ErrorBoundary.props.children.props.children;
    const beforeLoginRoutes = routes[0];
    expect(beforeLoginRoutes.key === 'Register').toBe(true);

    checkExpireTime.mockReturnValueOnce(true);
    const result2 = App({ history: {}, store });
    expect(clearStorages).toHaveBeenCalled();
    expect(window.location.reload).toHaveBeenCalled();
    expect(result2).toBe(null);
  });

  test('MainPages', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<MainPages />);
    expect(tree).toMatchSnapshot();
  });
});
