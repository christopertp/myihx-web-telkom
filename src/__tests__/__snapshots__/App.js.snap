// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App MainPages 1`] = `
<MainBase>
  <Switch>
    <Route
      exact={true}
      path="/about"
    />
    <Route
      exact={true}
      path={
        Array [
          "/activate-rewards",
          "/activate-rewards/:page",
        ]
      }
    />
    <Route
      exact={true}
      path="/bill"
    />
    <Route
      exact={true}
      path={
        Array [
          "/complete-profile",
          "/complete-profile/:page",
        ]
      }
    />
    <Route
      exact={true}
      path="/draft-contract"
    />
    <Route
      exact={true}
      path={
        Array [
          "/history",
          "/history/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/inbox",
          "/inbox/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/issues",
          "/issues/:id",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/linkaja",
          "/linkaja/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/manage-subscriptions",
          "/manage-subscriptions/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/manage-users",
          "/manage-users/:id",
          "/manage-users/:id/:action",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/notification-settings",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/installation",
          "/installation/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/order/:page",
          "/order/:page/:subpage",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/payment-:code/:method",
          "/payment-:code/:method/:detail",
          "/payment-:code/:method/:detail/:bank",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/personal-data",
          "/personal-data/:page",
          "/personal-data/:page/:subpage",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/profile",
          "/profile/:page",
          "/profile/:page/:subpage",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/promo/:name",
        ]
      }
    />
    <Route
      exact={true}
      path="/receipt/:period"
    />
    <Route
      exact={true}
      path={
        Array [
          "/rewards/:page",
          "/rewards/:page/:id",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/schedule-:type",
          "/schedule-:type/:page",
        ]
      }
    />
    <Route
      exact={true}
      path="/send-feedback"
    />
    <Route
      exact={true}
      path={
        Array [
          "/tmoney",
          "/tmoney/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/check-coverage",
          "/check-coverage/:page",
        ]
      }
    />
    <Route
      exact={true}
      path="/dummy-nps"
    />
    <Route
      exact={true}
      path={
        Array [
          "/help",
          "/help/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/shop/internet/package/category",
          "/shop/internet/package/category/:category",
        ]
      }
    />
    <Route
      exact={true}
      path="/shop/internet/package/:productId"
    />
    <Route
      exact={true}
      path="/search"
    />
    <Route
      exact={true}
      path={
        Array [
          "/shop/internet",
          "/shop/internet/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/shop/more",
          "/shop/more/:page",
        ]
      }
    />
    <Route
      exact={true}
      path={
        Array [
          "/",
          "/shop",
        ]
      }
    />
    <Route />
  </Switch>
</MainBase>
`;

exports[`src/App render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <Router
    history={Object {}}
  >
    <ErrorBoundary>
      <ContextProvider>
        <Switch>
          <Route
            exact={true}
            path={
              Array [
                "/forgot-password",
                "/forgot-password/:page",
              ]
            }
          />
          <Route
            component="login"
            exact={true}
            path={
              Array [
                "/login",
                "/login/:page",
              ]
            }
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </ContextProvider>
    </ErrorBoundary>
  </Router>
</Provider>
`;
